app/src/main/java/com/skyyo/samples/features/marqueeText/MarqueeText.kt

```diff
@@ -0,0 +1,206 @@
+import androidx.compose.ui.layout.SubcomposeLayout
+import androidx.compose.ui.layout.SubcomposeMeasureScope
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.TextUnit
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+
+enum class ScrollDirection { Forward, Backward }
+const val SPACING_NONE = 0
+const val SPACING_LARGE = 1
+const val SPACING_MEDIUM = 2
+const val SPACING_SMALL = 3
+
+private enum class MarqueeComponents { MainText, NextText, EdgesGradient }
+private data class TextLayoutInfo(val textWidth: Int, val containerWidth: Int)
+
+@Composable
+fun MarqueeText(
+    text: String,
+    modifier: Modifier = Modifier,
+    gradientEdgeColor: Color = Color.White,
+    scrollDirection: ScrollDirection = ScrollDirection.Forward,
+    scrollStartDelay: Int = 2000,
+    textEndsSpacing: Int = SPACING_LARGE,
+    edgeGradientWidth: Dp = 12.dp,
+    color: Color = Color.Unspecified,
+    fontSize: TextUnit = TextUnit.Unspecified,
+    fontStyle: FontStyle? = null,
+    fontWeight: FontWeight? = null,
+    fontFamily: FontFamily? = null,
+    letterSpacing: TextUnit = TextUnit.Unspecified,
+    textDecoration: TextDecoration? = null,
+    textAlign: TextAlign? = null,
+    lineHeight: TextUnit = TextUnit.Unspecified,
+    overflow: TextOverflow = TextOverflow.Clip,
+    softWrap: Boolean = true,
+    onTextLayout: (TextLayoutResult) -> Unit = {},
+    style: TextStyle = LocalTextStyle.current,
+) {
+    val createText = @Composable { localModifier: Modifier ->
+        Text(
+            text,
+            textAlign = textAlign,
+            modifier = localModifier,
+            color = color,
+            fontSize = fontSize,
+            fontStyle = fontStyle,
+            fontWeight = fontWeight,
+            fontFamily = fontFamily,
+            letterSpacing = letterSpacing,
+            textDecoration = textDecoration,
+            lineHeight = lineHeight,
+            overflow = overflow,
+            softWrap = softWrap,
+            maxLines = 1,
+            onTextLayout = onTextLayout,
+            style = style,
+        )
+    }
+    var mainTextOffset by remember { mutableStateOf(0) }
+    val textLayoutInfoState = remember { mutableStateOf<TextLayoutInfo?>(null) }
+
+    val scrollAnimationUpdateDelay = 1000L
+
+    LaunchedEffect(textLayoutInfoState.value) {
+        val textLayoutInfo = textLayoutInfoState.value ?: return@LaunchedEffect
+
+        val duration = 7500 * textLayoutInfo.textWidth / textLayoutInfo.containerWidth
```

Comment:
let's please move this duration as a param with a descriptive name, which can be provided outside, with default value of 7500